spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/channel/PostgresChannelMessageTableSubscriber.java

```diff
@@ -208,6 +217,29 @@ public synchronized void start() {
 				this.latch.countDown();
 			}
 		});
+
+		try {
+			if (!startingLatch.await(5, TimeUnit.SECONDS)) {
```

Comment:
To mitigate a race condition you can just swap subscription to this publisher in the test with its start call.
But I think I'm OK with this `startingLatch` since it give us a chance to ensure that we call `LISTEN` before any inserts into a table.

Reply:
My opinion is that having a latch gives a more deterministic behavior